letsencrypt/models/letsencrypt.py

```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+# © 2016 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import os
+import logging
+import urllib2
+import urlparse
+import subprocess
+import tempfile
+from openerp import _, api, models, exceptions
+from openerp.tools.config import _get_default_datadir
+from openerp.tools.misc import file_open
+
+
+DEFAULT_KEY_LENGTH = 4096
+_logger = logging.getLogger(__name__)
+
+
+class Letsencrypt(models.AbstractModel):
+    _name = 'letsencrypt'
+    _description = 'Abstract model providing functions for letsencrypt'
+
+    @api.model
+    def get_data_dir(self):
+        return os.path.join(_get_default_datadir(), 'letsencrypt')
```

Comment:
I test this module currently, and I encounter a bug: `_get_default_datadir` returns the default data directory of Odoo (`~/.local/share/Odoo`), but does not honor the `data_dir` configuration option.